tools/run_tests/xds_k8s_test_driver/framework/test_app/server_app.py

```diff
@@ -325,25 +323,22 @@ def run(self,
 
             pod_ip = pod.status.pod_ip
             rpc_host = None
+            rpc_port = test_port
             # Experimental, for local debugging.
             local_port = maintenance_port
             if self.debug_use_port_forwarding:
-                with KubernetesServerRunner._lock:
-                    local_port = maintenance_port + KubernetesServerRunner._server_port_forwarding_offset
-                    KubernetesServerRunner._server_port_forwarding_offset += 1
-                logger.info(
-                    'LOCAL DEV MODE: Enabling port forwarding to %s:%s using local port %s',
-                    pod_ip, maintenance_port, local_port)
-                self.port_forwarders.append(
-                    self.k8s_namespace.port_forward_pod(
-                        pod,
-                        remote_port=maintenance_port,
-                        local_port=local_port))
+                logger.info('LOCAL DEV MODE: Enabling port forwarding to %s:%s',
+                            pod_ip, maintenance_port)
+                local_port, port_forwarder = self.k8s_namespace.port_forward_pod(
+                    pod, remote_port=maintenance_port)
+                self.port_forwarders.append(port_forwarder)
+                self.local_forwarding_ports.append(local_port)
```

Comment:
Same here: unset `self.local_forwarding_ports` in `self.cleanup()`

Reply:
`local_forwarding_ports` removed, for the `PortForwarder` class.